src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -388,16 +387,49 @@ private static bool IsValidIssuer(string validIssuerTemplate, string tenantId, s
             if (string.IsNullOrEmpty(validIssuerTemplate))
                 return false;
 
-            if (validIssuerTemplate.Contains(TenantIdTemplate))
+            int indexOfTenantIdTemplate = validIssuerTemplate.IndexOf(TenantIdTemplate, StringComparison.Ordinal);
+            if (indexOfTenantIdTemplate >= 0)
             {
-                return validIssuerTemplate.Replace(TenantIdTemplate, tenantId) == actualIssuer;
+                return IssuersWithTemplatesAreEqual(validIssuerTemplate.AsSpan(), TenantIdTemplate.AsSpan(), indexOfTenantIdTemplate, actualIssuer.AsSpan(), tenantId.AsSpan());
             }
             else
             {
                 return validIssuerTemplate == actualIssuer;
             }
         }
 
+        /// <summary>
+        /// Compare two Issuers with templates without string allocations.
+        /// This function replaces: issuer1.Replace(issuer1Template, tenantId) == issuer2
+        /// Example:
+        ///     issuer1 = "https://login.microsoftonline.com/{tenantid}/v2.0"
```

Comment:
The naming throws me off a bit.
issuer1 is the value from metadata or TokenValidationParameters, it is actually the expected value, that may contain a value that may be substituted: {tenantid}. Maybe 'expectedIssuer' or 'issuerToMatch' are suggestions.

 issuer2 is the 'iss' value of the JWT. Could just be named 'issuer' or 'jwtIssuer'.

Reply:
Updated the naming, thank you for providing suggestions!